src/protocol/protocol.c

```diff
@@ -159,36 +158,23 @@ void PROTOCOL_Load(int no_dlg)
     fh = fopen2(&FontFAT, file, "r");
     //printf("Loading %s: %08lx\n", file, fh);
     if(! fh) {
+        LCD_SetFont(old_font);
         if(! no_dlg) {
             sprintf(tempstring, "Misisng protocol:\n%s", file);
             PAGE_ShowWarning(NULL, tempstring);
         }
-        LCD_SetFont(old_font);
         return;
     }
     setbuf(fh, 0);
-    int size = 0;
-    unsigned char buf[256];
-    int len;
-    char *ptr = (char *)loaded_protocol;
-    while(size < 4096) {
-        len = fread(buf, 1, 256, fh);
-        if(len) {
-            memcpy(ptr, buf, len);
-            ptr += len;
-        }
-        size += len;
-        if (len != 256)
-            break;
-    }
+    fread(loaded_protocol, 1, 4 * 1024, fh);
     fclose(fh);
```

Comment:
I think this is ok, but need to double check that all implementations of fread() can handle large reads.  there may be some FS issue here.  Probably not though.

Reply:
I confirm both pf_read and df_read support read more than more the sector size.